presto-hive/src/main/java/com/facebook/presto/hive/HivePartitioningHandle.java

```diff
@@ -49,21 +81,33 @@ public int getBucketCount()
     }
 
     @JsonProperty
-    public List<HiveType> getHiveTypes()
+    public Optional<List<HiveType>> getHiveTypes()
     {
         return hiveTypes;
     }
 
+    @JsonProperty
+    public Optional<List<Type>> getTypes()
+    {
+        return types;
+    }
+
     @JsonProperty
     public OptionalInt getMaxCompatibleBucketCount()
     {
         return maxCompatibleBucketCount;
     }
 
+    @JsonProperty
+    public HiveBucketProperty.BucketingType getBucketingType()
+    {
+        return bucketingType;
+    }
+
     @Override
     public String toString()
     {
-        return format("buckets=%s, hiveTypes=%s", bucketCount, hiveTypes);
+        return format("buckets=%s, hiveTypes=%s, types=%s, bucketingType=%s", bucketCount, hiveTypes, types, bucketingType);
```

Comment:
This is what is printed in the plan. In a current form it might be a little hard to read. How about we merge `hiveTypes` and `types` and print only one?

e.g.:

```
format("bucketFunctionType=%s, buckets=%s, types=%s", bucketingType, bucketCount, hiveTypes or types);
```